temba/msgs/models.py

```diff
@@ -378,15 +381,34 @@ def get_translated_text(self, contact, org=None):
         preferred_languages = self.get_preferred_languages(contact, org)
         return Language.get_localized_text(self.text, preferred_languages)
 
+    def get_translated_metadata(self, metadata, contact, org=None):
+        """
+        Gets the appropriate metadata translation for the given contact
+        """
+        preferred_languages = self.get_preferred_languages(contact, org)
+        language_metadata = []
+        for item in metadata.get('quick_replies'):
+            current_language = preferred_languages[0] if preferred_languages else None
```

Comment:
This shouldn't be in the loop.

Also should be using Language.get_localized_text() which is smart about how things backdown when languages are missing.